ConnectionFactory.php

```diff
@@ -45,8 +45,13 @@ public function createConnection(array $params, Configuration $config = null, Ev
             $this->initializeTypes();
         }
 
-        if (! isset($params['pdo']) && ! isset($params['charset'])) {
-            $wrapperClass = null;
+        $allowOverrideUrl = $params['override_url'] ?? false;
+        unset($params['override_url']);
+
+        if (! isset($params['pdo']) && (! isset($params['charset']) || $allowOverrideUrl)) {
+            $originalParams = $params;
+            $wrapperClass   = null;
```

Comment:
Why is this needed?

Reply:
`$wrapperClass`? it was already there.. `$originalParams` is needed so because `createConnection()` overwrites the `params[$key]` when it parses the `url`

Reply:
got it - the diff looks, unfortunately, bigger than it actually is :/